### PR TITLE
Implement debug execution from docker containers 

### PR DESCRIPTION
**Pull request by louneskmt**
This PR adds a trigger to execute the debug script and save the output.

Related PRs:
- https://github.com/getumbrel/umbrel-dashboard/pull/275 (main)
- https://github.com/getumbrel/umbrel-manager/pull/66

### CODE REVIEW

File: events/triggers/debug [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Function to upload the output to hastebin
+# Source: https://github.com/seejohnrun/haste-client#lightweight-alternative
+haste() { a=$(cat); curl -X POST -s -d "$a" https://hastebin.com/documents | awk -F '"' '{print "https://hastebin.com/"$4}'; }
+
+UMBREL_ROOT="$(readlink -f $(dirname "${BASH_SOURCE[0]}")/../..)"
+DEBUG_FILE="${UMBREL_ROOT}/scripts/debug"
+
+if [ ! -f ${DEBUG_FILE} ]; then
+  wget https://raw.githubusercontent.com/getumbrel/umbrel/master/scripts/debug -O ${DEBUG_FILE} && chmod +x ${DEBUG_FILE}
+fi
```
Comment (AaronDewes) [inline events/triggers/debug:12]: When can this condition be true? If the user deletes files, we shouldn't have to have code for that specific case.

File: events/triggers/debug [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Function to upload the output to hastebin
+# Source: https://github.com/seejohnrun/haste-client#lightweight-alternative
+haste() { a=$(cat); curl -X POST -s -d "$a" https://hastebin.com/documents | awk -F '"' '{print "https://hastebin.com/"$4}'; }
+
+UMBREL_ROOT="$(readlink -f $(dirname "${BASH_SOURCE[0]}")/../..)"
+DEBUG_FILE="${UMBREL_ROOT}/scripts/debug"
```
Comment (AaronDewes) [inline events/triggers/debug:8]: I think DEBUG_SCRIPT would be a better name, DEBUG_FILE sounds like a file the output of the script is stored in.

**Review by AaronDewes — CHANGES_REQUESTED** [review]

File: events/triggers/debug (hunk 1 above)
Comment (louneskmt) [reply events/triggers/debug:12]: For example if the user doesn't run the latest Umbrel version and thus the script is not included.

File: events/triggers/debug (hunk 2 above)
Comment (louneskmt) [reply events/triggers/debug:8]: Yeah you're right, gonna change this

File: events/triggers/debug (hunk 1 above)
Comment (AaronDewes) [reply events/triggers/debug:12]: But then this trigger wouldn't exist either.

Comment (louneskmt) [reply events/triggers/debug:12]: Yeah, seen like this 😂

File: events/triggers/debug [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Function to upload the output to hastebin
+# Source: https://github.com/seejohnrun/haste-client#lightweight-alternative
+haste() { a=$(cat); curl -X POST -s -d "$a" https://hastebin.com/documents | awk -F '"' '{print "https://hastebin.com/"$4}'; }
```
Comment (AaronDewes) [inline events/triggers/debug:5]:
```suggestion
# function to upload the output to our paste server
# Based on https://github.com/seejohnrun/haste-client#lightweight-alternative
upload() { a=$(cat); curl -X POST -s -d "$a" https://umbrel-paste.vercel.app/documents | awk -F '"' '{print "https://umbrel-paste.vercel.app/"$6}'; }
```

File: events/triggers/debug (hunk 1 above)
Comment (AaronDewes) [inline events/triggers/debug:12]:
```suggestion
```

File: events/triggers/debug [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+if [ ! -f ${DEBUG_FILE} ]; then
+  wget https://raw.githubusercontent.com/getumbrel/umbrel/master/scripts/debug -O ${DEBUG_FILE} && chmod +x ${DEBUG_FILE}
+fi
+
+cat <<EOF > "$UMBREL_ROOT"/statuses/debug-result.json
+{"status": "processing", "linkDebug": "", "linkDmesg": "", "output": ""}
+EOF
+
+debuglink=$(${DEBUG_FILE} | sed '/onion/d' | haste)
+dmesglink=$(dmesg | haste)
```
Comment (AaronDewes) [inline events/triggers/debug:19]:
```suggestion
debuglink=$(${UMBREL_ROOT}/scripts/debug | sed '/onion/d' | upload)
dmesglink=$(dmesg | upload)
```

File: events/triggers/debug (hunk 2 above)
Comment (AaronDewes) [inline events/triggers/debug:8]:
```suggestion
```

File: scripts/debug [hunk 5]
```diff
@@ -155,7 +155,7 @@ if [[ ! -z "${UMBREL_OS:-}" ]]; then
         echo "After you've finished that, boot your Raspberry Pi back up. If it still still doesn't work,"
         echo "you can contact us on Telegram (t.me/getumbrel) and share the output of this script."
         if [[ "${1}" == "--upload" ]]; then
-            echo "You can also share this link instead:"
+            echo "You can also share these links instead:"
```
Comment (AaronDewes) [inline scripts/debug:158]:
```suggestion
            echo "You can also share this link instead:"
```

This shouldn't be part of this PR

**Review by AaronDewes — CHANGES_REQUESTED** [review]
A few changes we spoke about before, as well as switching to the new server.

File: events/triggers/debug (hunk 1 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 589207a.

File: events/triggers/debug (hunk 2 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 589207a.

File: events/triggers/debug (hunk 3 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 589207a.

File: events/triggers/debug (hunk 1 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 589207a.

File: events/triggers/debug (hunk 4 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 589207a.

File: events/triggers/debug (hunk 2 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 589207a.

File: events/triggers/debug [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+debug_result=$(${DEBUG_SCRIPT} | sed '/onion/d')
+
+dmesglink=$(dmesg | upload)
+debuglink=$(echo "$debug_result" | upload)
+result64=$(echo "$debug_result" | base64 | tr -d '\n')
+
+cat <<EOF > "$UMBREL_ROOT"/statuses/debug-result.json
+{"status": "success", "linkDebug": "${debuglink}", "linkDmesg": "${dmesglink}", "output": "${result64}"}
+EOF
```
Comment (AaronDewes) [inline events/triggers/debug:22]:
```suggestion
debuglink=$(echo "${debug_result} === Umbrel-Paste split === $(dmesg)" | upload)
result64=$(echo "$debug_result" | base64 | tr -d '\n')

cat <<EOF > "$UMBREL_ROOT"/statuses/debug-result.json
{"status": "success", "linkDebug": "${debuglink}", "output": "${result64}"}
EOF
```

File: events/triggers/debug [hunk 7]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# function to upload the output to our paste server
+# Based on https://github.com/seejohnrun/haste-client#lightweight-alternative
+upload() { a=$(cat); curl -X POST -s -d "$a" https://umbrel-paste.vercel.app/documents | awk -F '"' '{print "https://umbrel-paste.vercel.app/"$6}'; }
```
Comment (AaronDewes) [inline events/triggers/debug:5]:
```suggestion

upload() {
  curl \
    --header "Content-Type: text/plain" \
    --request POST \
    --silent \
    --data-binary @- \
    https://debug.umbrel.tech/documents \
    | awk -F '"' '{print "https://debug.umbrel.tech/"$6}'
}
```

You can switch to the new URL here, fix the upload and make command more clear.

**Review by AaronDewes — CHANGES_REQUESTED** [review]
Needs to be updated for server v2

File: events/triggers/debug (hunk 6 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 1d131c5.

File: events/triggers/debug (hunk 7 above)
Comment (louneskmt) [reply events/triggers/debug]: Resolved in 1d131c5.

**Comment by lukechilds** [comment]
Thanks @louneskmt, awesome stuff!
